network/tcp.go

```diff
@@ -379,38 +378,37 @@ func (t *TCPListener) Listening() bool {
 
 // TCPHost implements the Host interface using TCP connections.
 type TCPHost struct {
-	addr  Address
 	suite Suite
+	sid   *ServerIdentity
 	*TCPListener
 }
 
 // NewTCPHost returns a new Host using TCP connection based type.
-func NewTCPHost(addr Address, s Suite) (*TCPHost, error) {
+func NewTCPHost(sid *ServerIdentity, s Suite) (*TCPHost, error) {
 	h := &TCPHost{
-		addr:  addr,
 		suite: s,
+		sid:   sid,
 	}
 	var err error
-	h.TCPListener, err = NewTCPListener(addr, s)
+	if sid.Address.ConnType() == TLS {
+		h.TCPListener, err = NewTLSListener(sid, s)
+	} else {
+		h.TCPListener, err = NewTCPListener(sid.Address, s)
```

Comment:
Maybe write a `deprecated plain TCP connection` on the log or stg ?

Reply:
They are not deprecated. If TCP was the right choice for an application, we can support it still.

Reply:
Should it not be deprecated ?  Since you implemented encrypted communication by keeping the same API and the same deployement workflow, we should encourage the use of TLS and start deprecating less secure communication transport.

Reply:
I think we should keep the TCP connections around - if it's just for easier debugging. So -1 for deprecation warning.